pkg/cluster/k8sres.go

```diff
@@ -1905,40 +1919,49 @@ func (c *Cluster) generateCloneEnvironment(description *acidv1.CloneDescription)
 func (c *Cluster) generateStandbyEnvironment(description *acidv1.StandbyDescription) []v1.EnvVar {
 	result := make([]v1.EnvVar, 0)
 
-	if description.S3WalPath == "" && description.GSWalPath == "" {
+	if description.StandbyMethod == "" {
 		return nil
 	}
 
-	if description.S3WalPath != "" {
-		// standby with S3, find out the bucket to setup standby
-		msg := "Standby from S3 bucket using custom parsed S3WalPath from the manifest %s "
-		c.logger.Infof(msg, description.S3WalPath)
-
-		result = append(result, v1.EnvVar{
-			Name:  "STANDBY_WALE_S3_PREFIX",
-			Value: description.S3WalPath,
-		})
-	} else if description.GSWalPath != "" {
-		msg := "Standby from GS bucket using custom parsed GSWalPath from the manifest %s "
-		c.logger.Infof(msg, description.GSWalPath)
-
-		envs := []v1.EnvVar{
-			{
-				Name:  "STANDBY_WALE_GS_PREFIX",
-				Value: description.GSWalPath,
-			},
-			{
-				Name:  "STANDBY_GOOGLE_APPLICATION_CREDENTIALS",
-				Value: c.OpConfig.GCPCredentials,
-			},
-		}
-		result = append(result, envs...)
+	if description.StandbyMethod == "s3_wal" || description.StandbyMethod == "gs_wal" {
+  	if description.S3WalPath != "" {
+  		// standby with S3, find out the bucket to setup standby
+	  	msg := "Standby from S3 bucket using custom parsed S3WalPath from the manifest %s "
+		  c.logger.Infof(msg, description.S3WalPath)
+
+	  	result = append(result, v1.EnvVar{
+	  		Name:  "STANDBY_WALE_S3_PREFIX",
+		  	Value: description.S3WalPath,
+		  })
+  	} else if description.GSWalPath != "" {
+	  	msg := "Standby from GS bucket using custom parsed GSWalPath from the manifest %s "
+	  	c.logger.Infof(msg, description.GSWalPath)
+
+	  	envs := []v1.EnvVar{
+	  		{
+	  			Name:  "STANDBY_WALE_GS_PREFIX",
+	  			Value: description.GSWalPath,
+	  		},
+		  	{
+			  	Name:  "STANDBY_GOOGLE_APPLICATION_CREDENTIALS",
+				  Value: c.OpConfig.GCPCredentials,
+			  },
+		  }
+		  result = append(result, envs...)
+  	}
+		result = append(result, v1.EnvVar{Name: "STANDBY_METHOD", Value: "STANDBY_WITH_WALE"})
+		result = append(result, v1.EnvVar{Name: "STANDBY_WAL_BUCKET_SCOPE_PREFIX", Value: ""})
+	} else if description.StandbyMethod == "streaming_host" {
+		// standby with streaming replication from standby host
+		msg := "Standby using streaming replication from standby host: %s "
+		c.logger.Infof(msg, description.StandbyHost)
+
+		result = append(result, v1.EnvVar{Name: "STANDBY_HOST", Value: description.StandbyHost})
+		result = append(result, v1.EnvVar{Name: "STANDBY_PORT", Value: description.StandbyPort})
```

Comment:
I think STANDBY_PORT can be omitted when it's not set. Patroni will default to 5432.